sw/device/tock/chips/opentitan_common/src/uart.rs

```diff
@@ -246,15 +175,36 @@ impl hil::uart::Uart<'a> for Uart<'a> {}
 
 impl hil::uart::Configure for Uart<'a> {
     fn configure(&self, params: hil::uart::Parameters) -> ReturnCode {
-        let regs = self.registers;
-        // We can set the baud rate.
-        self.set_baud_rate(params.baud_rate);
+        let base_addr: mmio_region_t = mmio_region_t {
+            base: self.base_addr as *mut c_void,
+        };
+
+        let mut config: dif_uart_config_t = dif_uart_config_t {
+            baudrate: params.baud_rate,
+            clk_freq_hz: self.clock_frequency,
+            parity_enable: kDifUartDisable,
+            parity: kDifUartParityEven,
+        };
+
+        let config_p: *mut dif_uart_config_t = &mut config;
+
+        let mut uart: dif_uart_t = Default::default();
+        let uart_p: *mut dif_uart_t = &mut uart;
+
+        unsafe {
+            match dif_uart_init(base_addr, config_p, uart_p) {
+                kDifUartOk => self.dif_uart.set(uart),
+                _ => return ReturnCode::FAIL,
+            }
+        }
```

Comment:
I think this points out that we need to separate the DIF's `_init` and `_config` for some of these peripherals, and that `_init` should not require a configuration struct. Then instead of storing a base address, you can just store a non-null `dif_uart_t`.

Reply:
Sounds good, I think that is what we are converging to for all DIFs anyway?

Reply:
As discussed offline with @lenary , we think this should be a follow-up, as it consists of changes to DIF UART API.